Initial cut at transport layer security text... (#19)
uptane/deployment-considerations: customizations.md

However, in order for an implementation to be Uptane-compliant, no ECU can cause another ECU to install an image without performing either full or partial verification of metadata. This is done in order to prevent attackers from being able to bypass Uptane, and thus execute arbitrary software attacks. Thus, in an Uptane-compliant implementation, an ECU performs either full or partial verification of metadata and images before installing any image, regardless of how the metadata and images were transmitted to the ECU.

## Using Uptane with transport security

> Uptane's security guarantees are designed to operate even in the face of a network attacker.  This includes situations where there either is no transport security or where that security is compromised by an attacker.  In this case, Uptane's security loss is that an attacker may disrupt communication between the vehicle and the OEM (e.g., by jamming the signal or dropping packets).  However, malicious packages cannot be installed, mix-and-match attacks are not possible, etc.  This is similar to how a network attacker without key material can cause a TLS connection to fail to connect or disconnect, but cannot compromise the integrity or confidentiality of the connection.

Author: ```suggestion
Uptane is designed to retain strong security guarantees even in the face of a network attacker.  This includes situations where there either is no transport security or where the transport security is compromised by an attacker.  In this case, Uptane's security loss is that an attacker may disrupt communication between the vehicle and the OEM (e.g., by jamming the signal or dropping packets).  However, malicious packages cannot be installed, mix-and-match attacks are not possible, etc.  This is similar to how a network attacker without key material can cause a TLS connection to fail to connect or disconnect, but cannot compromise the integrity or confidentiality of the connection.
```

Uptane's security is orthogonal to security systems at other network layers, such as transport security, data link security, etc.  However, there are several reasons why a party may wish to use a security system at the transport layer in coordination with Uptane:

- If a security system at the transport layer is already deployed for other services or is effectively free to deploy, there is little reason not to use it.  For example, it may be beneficial to have authentication provided by a common system for all services in a vehicle, so this may be considered desirable from a uniformity standpoint.

- Regulation may require or suggest that security at the transport layer is provided.  Hence it may be required for non-technical reasons.

- It does not weaken Uptane's security to use Uptane over a transport layer security system.  If the cost is low, then this can be viewed in some situations as adding defense-in-depth, especially if the security system can improve detection, mitigation, or reporting of network disruptions.

> 

Author: ```suggestion
- Security at the transport layer may offer forensic proof of origin and destination (when strong mutual authentication is used).  This may be necessary for compliance to OTA update standards and various current draft
regulations.
```

However, in order for an implementation to be Uptane-compliant, no ECU can cause another ECU to install an image without performing either full or partial verification of metadata. This is done in order to prevent attackers from being able to bypass Uptane, and thus execute arbitrary software attacks. Thus, in an Uptane-compliant implementation, an ECU performs either full or partial verification of metadata and images before installing any image, regardless of how the metadata and images were transmitted to the ECU.

## Using Uptane with transport security

> Uptane is designed to retain strong security guarantees even in the face of a network attacker.  This includes situations where there either is no transport security or where the transport security is compromised by an attacker.  In this case, Uptane's security loss is that an attacker may disrupt communication between the vehicle and the OEM (e.g., by jamming the signal or dropping packets).  However, malicious packages cannot be installed, mix-and-match attacks are not possible, etc.  This is similar to how a network attacker without key material can cause a TLS connection to fail to connect or disconnect, but cannot compromise the integrity or confidentiality of the connection.

Reviewer: "Uptane's security loss" sounds a bit odd to me. Can we find a better way to phrase that?

Reviewer 2: I agree with Patrick. I suggest rephrasing the beginning of that third sentence as follows:

"Should this occur, Uptane may not be able to prevent an attacker from disrupting communication between the vehicle and the OEM (e.g...."

In the last sentence in this paragraph, the phrase "key material" strikes me as odd. Can't we just say "...a network attacker who has not compromised a key can cause..."

Reviewer 2: The unfinished bits at the end of my comment can be ignored. I think I based this on an older message and then just forgot to cut out what wasn't needed.
The three suggested edits are in the message,

Lois

Author: Okay, I've added the suggestions mentioned.  (Lois, next time please do that instead of listing them only in the comment.)

Can someone outside NYU please review / merge if this is okay?

Reviewer 2: Yes. Will do,

> Uptane is designed to retain strong security guarantees even in the face of a network attacker.  This includes situations where there either is no transport security or where the transport security is compromised by an attacker.  In this case, Uptane's security loss is that an attacker may disrupt communication between the vehicle and the OEM (e.g., by jamming the signal or dropping packets).  However, malicious packages cannot be installed, mix-and-match attacks are not possible, etc.  This is similar to how a network attacker without key material can cause a TLS connection to fail to connect or disconnect, but cannot compromise the integrity or confidentiality of the connection.

Author: ```suggestion
Uptane is designed to retain strong security guarantees even in the face of a network attacker.  This includes situations where there either is no transport security or where the transport security is compromised by an attacker.  Should this occur, Uptane may not be able to prevent an attacker from disrupting communication between the vehicle and the OEM (e.g., by jamming the signal or dropping packets).  However, malicious packages cannot be installed, mix-and-match attacks are not possible, etc.  This is similar to how a network attacker who has not compromised a key can cause a TLS connection to fail to connect or disconnect, but cannot compromise the integrity or confidentiality of the connection.
```

> Uptane is designed to retain strong security guarantees even in the face of a network attacker.  This includes situations where there either is no transport security or where the transport security is compromised by an attacker.  Should this occur, Uptane may not be able to prevent an attacker from disrupting communication between the vehicle and the OEM (e.g., by jamming the signal or dropping packets).  However, malicious packages cannot be installed, mix-and-match attacks are not possible, etc.  This is similar to how a network attacker who has not compromised a key can cause a TLS connection to fail to connect or disconnect, but cannot compromise the integrity or confidentiality of the connection.

Author: ```suggestion
Uptane is designed to retain strong security guarantees even in the face of a network attacker.  This includes situations where there either is no transport security or where the transport security is compromised by an attacker.  Should this occur, Uptane may not be able to prevent an attacker from disrupting communication between the vehicle and the OEM (e.g., by jamming the signal or dropping packets).  However, malicious packages cannot be installed, mix-and-match attacks are not possible, etc.  This is similar to how a network attacker who has not compromised a key can cause a TLS connection to fail to connect or disconnect (e.g., by dropping network packets), but cannot compromise the integrity or confidentiality of the connection.
```

Uptane's security is orthogonal to security systems at other network layers, such as transport security, data link security, etc.  However, there are several reasons why a party may wish to use a security system at the transport layer in coordination with Uptane:

> - If a security system at the transport layer is already deployed for other services or is effectively free to deploy, there is little reason not to use it.  For example, it may be beneficial to have authentication provided by a common system for all services in a vehicle, so this may be considered desirable from a uniformity standpoint.

Author: ```suggestion
- If a security system at the transport layer is already deployed for other services or is effectively free to deploy, there is little reason not to use it.  For example, it may be beneficial to have authentication provided for all services in a vehicle by a common system.
```

> - Regulation may require or suggest that security at the transport layer is provided.  Hence it may be required for non-technical reasons.

Author: ```suggestion
- Regulation may require or recommend that security at the transport layer is provided.  Hence it may be required for non-technical reasons.
```

> - Regulation may require or recommend that security at the transport layer is provided.  Hence it may be required for non-technical reasons.

Reviewer: ```suggestion
- Transport layer security may be required for non-technical reasons, such as regulations that require or recommend it.
```

> - Transport layer security may be required for non-technical reasons, such as regulations that require or recommend it.

Author: ```suggestion
- Regulations may require or recommend that security be provided at the transport layer. Hence, a secure transport system may be required for non-technical reasons.
```

> - Use of transport layer security adds defense-in-depth, a security best practice, to the extent to which the transport layer security system can improve the detection, mitigation, or reporting of network disruptions.

Author: ```suggestion
- Using Uptane over a transport layer security system does not weaken its own security properties. If the cost is low, then this may be viewed as adding defense-in-depth, especially if the security system can improve detection, mitigation, or reporting of network disruptions.
```

> - - Use of transport layer security adds defense-in-depth, a security best practice, to the extent to which the transport layer security system can improve the detection, mitigation, or reporting of network disruptions.

Reviewer: I also think this sentence could be made clearer, perhaps by splitting it in two. My initial suggestion here is just for formatting.
```suggestion
- Use of transport layer security adds defense-in-depth, a security best practice, to the extent to which the transport layer security system can improve the detection, mitigation, or reporting of network disruptions.
```

> - It does not weaken Uptane's security to use Uptane over a transport layer security system.  If the cost is low, then this can be viewed in some situations as adding defense-in-depth, especially if the security system can improve detection, mitigation, or reporting of network disruptions.

Author: @iramcdonald edit:
```suggestion
- Use of transport layer security adds defense-in-depth, a security best practice.
```

Author: I understand the goal here, but I feel like we're missing part of the point.  Defense in depth really works when the systems are likely to be independently compromised or when one is compromised the other could compensate.  (E.g. two network firewalls or a host firewall + a network firewall.)  I'm not really understanding a scenario where this happens here.  How would a transport layer security system know that things at the Uptane layer are incorrect?  Wouldn't an attacker that compromises the director and can be a MITM and has other keys also almost certainly have the means to push something out through whatever transport layer security solution is in place?  

As such, I'd suggest the following compromise text:

```suggestion
- - Use of transport layer security adds defense-in-depth, a security best practice, to the extent to which the transport layer security system can improve the detection, mitigation, or reporting of network disruptions.
```

(I'm very open to more suggestions / follow on edits...)

> - Security at the transport layer may offer forensic proof of origin and destination (when strong mutual authentication is used).  This may be necessary for compliance to OTA update standards and various current draft regulations.

Author: @iramcdonald edit:
```suggestion
- Security at the transport layer provides forensic proof of origin and destination (when strong mutual authentication is used).  This may be necessary for compliance to OTA update standards and various current draft regulations.
```
